File: languages/cpp/01-foundations/operators-and-expressions/example/main.cpp
// Module focus: Combining values through expressions and readable calculations.
// Why it matters: the example makes it possible to build expressions with correct precedence
// and explicit intent before the learner tackles the exercises.

#include <iostream>
using namespace std;

// Fixed inputs make the consequence of integer division when a decimal result is expected visible
// and repeatable.
int main() {
    // These values exercise the normal path before the exercises vary the documented boundaries.
    const int a = 17;
    const int b = 5;

    // The printed result shows whether the program can distinguish arithmetic, comparison, and
    // logical operations.
    cout << "a = " << a << ", b = " << b << "\n\n";
    cout << "a + b = " << (a + b) << '\n';
    cout << "a - b = " << (a - b) << '\n';
    cout << "a * b = " << (a * b) << '\n';
    cout << "a / b (integer division) = " << (a / b) << '\n';
    cout << "a % b = " << (a % b) << '\n';

    const double preciseDivision = static_cast<double>(a) / b;
    cout << "a / b (double division) = " << preciseDivision << "\n\n";

    const bool isPositive = (a > 0);
    const bool isEven = (a % 2 == 0);
    const bool passesRule = isPositive && !isEven;

    cout << "isPositive: " << (isPositive ? "true" : "false") << '\n';
    cout << "isEven: " << (isEven ? "true" : "false") << '\n';
    cout << "passesRule (positive and odd): " << (passesRule ? "true" : "false") << '\n';

    return 0;
}

File: languages/java/01-foundations/operators-and-expressions/example/Main.java
// Module focus: Combining values through expressions and readable calculations.
// Why it matters: the example makes it possible to build expressions with correct precedence and
// explicit intent before the learner tackles the exercises.

import java.util.Locale;

public class Main {
    public static void main(String[] args) {
        Locale.setDefault(Locale.US);

        // Parentheses make the order of operations visible before Java applies precedence.
        double subtotal = 120.0;
        double discountRate = 0.15;
        double taxRate = 0.08;

        double discount = subtotal * discountRate;
        double taxable = subtotal - discount;
        double tax = taxable * taxRate;
        double total = taxable + tax;

        // Report each calculated amount with fixed decimals for deterministic output.
        System.out.printf("Discount: %.2f%n", discount);
        System.out.printf("Tax: %.2f%n", tax);
        System.out.printf("Total: %.2f%n", total);
    }
}
